x-pack/plugins/beats_management/public/components/table/table.tsx

```diff
@@ -78,16 +81,26 @@ export class Table extends React.Component<TableProps, TableState> {
   };
 
   public actionHandler = (action: AssignmentActionType, payload?: any): void => {
+    const selectedIds = this.state.selection.map((item: any) => item.id);
+
+    const selectedItems: any[] = [];
+    selectedIds.forEach((id: any) => {
+      const item = this.props.items.list.find(b => b.id === id);
+      if (item) {
+        selectedItems.push(item);
+      }
+    });
+
     if (this.props.actionHandler) {
-      this.props.actionHandler(action, payload);
+      this.props.actionHandler(action, payload, selectedItems);
     }
   };
 
   public render() {
     const { actionData, actions, hideTableControls, items, kueryBarProps, type } = this.props;
 
     const pagination = {
-      pageIndex: this.state.pageIndex,
+      pageIndex: this.props.items.page !== -1 ? this.props.items.page : this.state.pageIndex,
```

Comment:
Mmmmm maybe? You tell me. Because ES pages start at 0, I used -1 to be a shortcut to say unpaged. That number won’t ever really change. This value carries through to the REST api when making requests

Reply:
I'm fine with not using `const`s if you prefer.

Reply:
Yea I think not here